### PR TITLE
Add Productivity Tracking

### PR DESCRIPTION
**Pull request by jeffshek**

### CODE REVIEW

File: apis/betterself/v1/events/serializers.py [hunk 1]
```diff
@@ -47,3 +55,36 @@ class SupplementEventReadOnlySerializer(serializers.Serializer):
...
     uuid = serializers.UUIDField()
+
+
+class ProductivityLogReadSerializer(serializers.Serializer):
+    very_productive_time_minutes = serializers.IntegerField(required=False)
+    productive_time_minutes = serializers.IntegerField(required=False)
+    neutral_time_minutes = serializers.IntegerField(required=False)
+    distracting_time_minutes = serializers.IntegerField(required=False)
+    very_distracting_time_minutes = serializers.IntegerField(required=False)
+    day = serializers.DateField()
```
Comment (jeffshek) [inline apis/betterself/v1/events/serializers.py]: See how other APIs illustrate date/day, day seems stupid

File: apis/betterself/v1/events/tests.py [hunk 2]
```diff
@@ -96,3 +114,30 @@ def test_uniqueness_on_post_requests(self):
 
         self.assertTrue(amount_of_events_post_request > amount_of_events)
         self.assertEqual(amount_of_events_post_request_repeated, amount_of_events_post_request)
+
+
+class TestProductivityLogViews(BaseAPIv1Tests, GetRequestsTestsMixin, PostRequestsTestsMixin):
+    # python manage.py test apis.betterself.v1.events.tests.TestProductivityLogViews
+    TEST_MODEL = DailyProductivityLog
+
+    def setUp(self):
```
Comment (jeffshek) [inline apis/betterself/v1/events/tests.py]: This is missing a bunch of tests that probably should be added. IE. No edge cases were done.

File: apis/betterself/v1/utils/views.py [hunk 3]
```diff
@@ -26,3 +26,15 @@ def get_queryset(self):
 class BaseGenericListCreateAPIViewV1(ListCreateAPIView, UserQuerysetFilterMixin):
     def get_queryset(self):
         return self._get_queryset()
+
+
+class ReadOrWriteViewInterfaceV1(object):
```
Comment (jeffshek) [inline apis/betterself/v1/utils/views.py]: See if there are other mixin/interfaces that you've done with this pattern that you might be able to merge.
